src/emqx_channel.erl

```diff
@@ -1698,7 +1704,16 @@ parse_topic_filters(TopicFilters) ->
     lists:map(fun emqx_topic:parse/1, TopicFilters).
 
 %%--------------------------------------------------------------------
-%% Ensure disconnected
+%% Maybe & Ensure disconnected
+
+ensure_disconnected(connected, Reason, Channel)
+  when Reason =:= discarded orelse Reason =:= takeovered ->
```

Comment:
please add a comment here to explain that `takeovered` is a typo we can not fix due to backward compatibility reason.